### PR TITLE
chore(box): convert to tests to rtl (FE-6557)

### PR DESCRIPTION
**Pull request by nineteen88**
### Proposed behaviour

- convert tests to RTL
- remove unnecessary style tests

### Current behaviour

- tests are currently written in Enzyme

### Checklist

- [x] Commits follow our style guide
- [x] Unit tests added or updated if required

### CODE REVIEW

File: src/components/box/box.test.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  testStyledSystemLayout,
+  testStyledSystemFlexBox,
+  testStyledSystemGrid,
+  testStyledSystemPosition,
+} from "../../__spec_helper__/test-utils";
+import Box from "./box.component";
+import Logger from "../../__internal__/utils/logger";
+
+// mock Logger.deprecate so that no console warnings occur while running the tests
+const loggerSpy = jest.spyOn(Logger, "deprecate");
```
Comment (Parsium) [inline src/components/box/box.test.tsx:17]: **suggestion**: what do you think about removing this mock, so any legitimate warnings that the tests throw that need addressing aren't masked?

Comment (edleeks87) [reply src/components/box/box.test.tsx:17]: I think it's worth trying to mock it for the specific test we want to ignore the console for, yes there's a small chance that specific one will have another legitimate console warning it's not likely a huge issue and removing the mock altogether means we'll have a lot of superfluous console logging polluting the output when we run our tests

Comment (edleeks87) [reply src/components/box/box.test.tsx:17]: Actually, is there any deprecation warnings we need to mock here?

Comment (nineteen88) [reply src/components/box/box.test.tsx:17]: I guess that's actually hangover from when we had deprecation warnings on Box. So I'm happy to remove it.

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**